curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/63ff5850cd098f1796b5e9fa.md

```diff
@@ -0,0 +1,143 @@
+---
+id: 63ff5850cd098f1796b5e9fa
+title: Step 3
+challengeType: 0
+dashedName: step-3
+---
+
+# --description--
+
+The `fetch()` method returns a `Promise`, which is  placeholder object that either resolves to a `Response` object, or is rejected. You can use the `.then()` method to access the `Promise` if it fulfils and the `.catch()` method to handle errors when it rejects.
+
+You can attach a `.then()` to the `fetch()` method this way:
+
+```js
+fetch("sample-url-goes-here")
+  .then((res) => console.log(res))
+```
+
+Chain the `.then()` method to your fetch call. Inside the `.then()` method, add a callback function with a `res` parameter, then log the `res` to the console so you can see the `Response` object. Open your console and expand the `Response` object to see what it contains.
+
+Again, don't terminate the code with a semi-colon yet.
+
+**Note**: You can open up the built-in freeCodeCamp console or your browser's console to see the data you're getting. Again, don't terminate your code with a semi-colon yet.
```

Comment:
I tried to shorten this a bit, and remove concepts that will be taught later:

``````suggestion
The `fetch()` method returns a `Promise`, which is a placeholder object that will either be fulfilled if your request is successful, or rejected if your request is unsuccessful.

If the `Promise` is fulfilled, it resolves to a `Response` object, and you can use the `.then()` method to access the `Response`.

Here's how you can chain `.then()` to the `fetch()` method:

```js
fetch("sample-url-goes-here")
  .then((res) => console.log(res))
```

Chain the `.then()` method to your `fetch` call. Inside the `.then()` method, add a callback function with a `res` parameter, then log the `res` to the console so you can see the `Response` object. Open your console and expand the `Response` object to see what it contains.

Again, don't terminate the code with a semi-colon yet.

**Note**: You can open up the built-in freeCodeCamp console or your browser's console to see the data you're getting.
``````